ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -122,15 +122,17 @@ def _peer_groups(self) -> list | None:
                 peer_group_config = {"remote_as": self.shared_utils.bgp_as}
                 if self.shared_utils.wan_role:
                     # WAN OVERLAY peer group
-                    peer_group_config["ttl_maximum_hops"] = self.shared_utils.bgp_peer_groups["wan_overlay_peers"]["ttl_maximum_hops"]
-                    if self.shared_utils.wan_role == "server":
-                        peer_group_config["route_reflector_client"] = True
-                    peer_groups.append(
+                    peer_group_config.update(
```

Comment:
This triggers a reordering of structured_config, which is not necessarily bad but can we keep maybe the previsou approach to keep the name of the peer group first rather than the AS?

```python
                peer_group_config = {"remote_as": self.shared_utils.bgp_as}
                if self.shared_utils.wan_role:
                    # WAN OVERLAY peer group
                    peer_group_config["ttl_maximum_hops"] = self.shared_utils.bgp_peer_groups["wan_overlay_peers"]["ttl_maximum_hops"]
                    if self.shared_utils.wan_role == "server":
                        peer_group_config["route_reflector_client"] = True
                    if bfd_timers := self._generate_bfd_timers("wan_overlay_peers"):
                        peer_group_config["bfd_timers"] = bfd_timers
                    
                    peer_groups.append(
                        {
                            **self._generate_base_peer_group("wan", "wan_overlay_peers", update_source=self.shared_utils.vtep_loopback),
                            **peer_group_config,
                        }
                    )
```